apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Logger } from '@nestjs/common';
+
+import { getRedisInstance, getRedisProviderConfig, IRedisProviderConfig, Redis } from './redis-provider';
+
+import { ApiException } from '../../exceptions/api.exception';
+
+const LOG_CONTEXT = 'InMemoryCluster';
+
+export type InMemoryProviderClient = Redis;
+type InMemoryProviderConfig = IRedisProviderConfig;
+
+export class InMemoryProviderService {
```

Comment:
One very little thing here regarding naming, maybe we can consider a different name for it, since initially I was sure that this service is an inMemory provider for testing purposes (instead of testing against a real db)

I know we want something more general than redis here, don't have a better unfortuently. Just want to point this, not necessarily a problem.

Reply:
If we move away from in-memory databases we can easily do a different adapter service of other kind and just plug the dependant services (Cache, DistributedLock). I find this and other db adapter services to be able to live parallelly.